### ISSUE TITLE
MongoDB / Mongoose ESM

### ISSUE THREAD

**Issue by akreienbring** [grahamjenson/ger#39]
Hello, did you ever think about a MongoDB / Mongoose storage manager? 

I'm currently thinking about using GER but  switching over to RethinkDB is a little hurdle. Especially as RethinkDB is (so far) not available for my Windows System.

**Comment by grahamjenson** [comment]
Hey,

There is a well tested Event Store Manager (ESM) API that you could implement using MongoDB or Mongoose if you wanted. The easiest way to do this is to read what the in memory ESM is doing and try replicate that in a class using Mongo, then running the tests to see what passes and what fails.

If you submit a pull request with a Mongo ESM that passes all the tests I will merge it :) Having a Mongo ESM would be pretty cool too

**Comment by muneerulhudha** [comment]
Hello, I was wondering if a MongoDB ESM was built for this?

**Comment by grahamjenson** [comment]
No, and I have no plans in the near future to implement one. I have made GER in such a way that you could implement one without a PR as a seperate npm package. The interface required has many tests and is well defined :)
